File: 199-BinaryTreeRightSideView/199-BinaryTreeRightSideView.java
// Last updated: 2/27/2026, 9:37:03 PM
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<Integer> rightSideView(TreeNode root) {
        Queue<TreeNode> queue = new LinkedList<TreeNode>();

        List<Integer> ans = new ArrayList<>();

        if(root == null) return ans;

        queue.offer(root);

        while (!queue.isEmpty()) {
            int levelSize = queue.size();

            for (int i = 1; i <= levelSize; i++) {
                TreeNode curr = queue.poll();

                if(i == levelSize) ans.add(curr.val);

                if (curr.left != null) queue.offer(curr.left);
                if (curr.right != null) queue.offer(curr.right);
            }

        }

        return ans;
    }
}

File: 199-BinaryTreeRightSideView/199-BinaryTreeRightSideView.py
# Last updated: 7/12/2026, 6:21:30 PM
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def rightSideView(self, root: Optional[TreeNode]) -> List[int]:
        if not root: return []

        q = deque([root])
        ans = []

        while q:
            size = len(q)
            for i in range(size):
                curr = q.popleft()

                if curr.left: q.append(curr.left)
                if curr.right: q.append(curr.right)

                if i == size - 1:
                    ans.append(curr.val)
        return ans
                







        